### PR TITLE
Change docker storage path

### PR DESCRIPTION
**Pull request by xatier**
See issue [#1475](https://github.com/tootsuite/mastodon/issues/1475)

### CODE REVIEW

File: Running-Mastodon/Production-guide.md [hunk 1]
```diff
@@ -101,6 +101,31 @@ server {
 }
 ```
 
+## Docker storage
+
+
+It suggested to move docker storage to another path (`dockerd -g $PATH`) instead of `/var/lib/docker` by default. This can help from the server accidentally run out of disk space.
```
Comment (wxcafe) [inline Running-Mastodon/Production-guide.md]: running* out of disk space

Comment (ProgVal) [inline Running-Mastodon/Production-guide.md]: It is suggested *


This can prevent the server from accidentally running *

**Comment by codl** [comment]
I'm not sure what this achieves, as long as you're using named volumes like the default `docker-compose.yml` recommends then all user-created data (uploads, db, redis db) will end up wherever you have installed mastodon, and the only things that will end up in `/var/lib/docker` are the various boilerplate system files found in the images, and mastodon itself. I don't know that "a system where `/var` is on a disk that doesn't have enough space for a few docker images" is a common enough situation to include it in the documentation

**Comment by wxcafe** [comment]
@xatier do you agree with that last comment?

**Comment by Wonderfall** [comment]
I may add you shouldn't use `/etc/systemd/system/docker.service.d/docker-storage.conf` but `/etc/docker/daemon.json` instead : https://docs.docker.com/engine/reference/commandline/dockerd/#options

**Comment by xatier** [comment]
@wxcafe I would say no. Still, not everyone can afford a hug disk on his/her servers. Especially on a typical small VPS instance, 20 or 30GB SSD configuration is common. Plus, the base image size for mastodon is already nearly 1GB anyway (700MB on my system). Having a suggestion in the document doesn't hurt anyone.

@Wonderfall I guess it depends on how you run docker on your system? Using a drop-in snippet in systemd is a common way to override the options from `docker.service` unit file.

[1] https://www.freedesktop.org/software/systemd/man/systemd.unit.html

**Comment by Wonderfall** [comment]
I know, but using `/etc/docker/daemon.json` is the official way for Docker. On top of that, an update of docker-engine will overwrite the previous `docker.service`.

**Comment by wxcafe** [comment]
I don't see how moving the data out of /var and into, for example, /usr/, solves the problem of small disks. It solves the problem of /var being a small partition, but not the problem of the disk itself being small.

**Comment by xatier** [comment]
@wxcafe yup, saying about partition would be a more precise in this case.

**Comment by wxcafe** [comment]
So, as said above, most people nowadays don't have a separate partition for /var, and you should probably not have a separate partition for /home either on a server.

**Comment by xatier** [comment]
@wxcafe why not?

**Comment by wxcafe** [comment]
Exactly because of this. You're not reinstalling the system on your server often (you're probably not reinstalling ever), and you have backups (right?) if you have to do so. The only thing having separate partitions gives you is having to manage partition sizes and being short on space in one while having too much in another one.

**Comment by xatier** [comment]
It's not about either reinstalling the system or having *too much* space on another partition. People plan partitions regarding how they *expect the space needed* for different purpose when they install their systems. My intention was just providing a note for those who might need to move the path to a proper place.

If you guys do feel like this is not really a common case then please close this issue and provide recommended disk space required [here](https://github.com/tootsuite/documentation/blob/master/Running-Mastodon/Resources-needed.md).

Trying to shrink the base docker image would also be a great idea. `gargron/mastodon` has about 150MB Ruby gems 360MB node packages = 0.5GB.

Given unlimited resource,  running out of disk space won't happen anyway.

**Comment by wxcafe** [comment]
having a single / partition means you don't have to worry about this tho

**Comment by wxcafe** [comment]
I'll add recommended disk space in the install guide

**Pull request closed without merging**